src/shogun/machine/Machine.cpp

```diff
@@ -176,3 +176,21 @@ std::shared_ptr<LatentLabels> Machine::apply_latent(std::shared_ptr<Features> da
 	error("This machine does not support apply_latent()");
 	return NULL;
 }
+
+std::shared_ptr<Machine> Machine::fit(
+    const std::shared_ptr<Features>& data, const std::shared_ptr<Labels>& lab)
+{
+	require(data, "Features not set!");
```

Comment:
pls have  a look at the train method of Machine ... there needs to me more stuff in here.
I also don't see why we would need to add fit_machine, rather than re-use train machine?

Reply:
when the whole refactor is done, `fit_machine` should be the same as `train_machine`, but if i change the parameters of `train_machine`,  it will be too much work to refactor everything in one pr, so `fit_machine` is just a temporary method.

Reply:
But I think that this non parametric machine can be done before this big refactor, thus you don’t need these methods in here yet.

Reply:
Non-parametric machines also needs to be refactored, right?if my understand is right, we want to change `train_machine(shared_ptr<Features>)` to `fit(shared_ptr<Features>, shared_ptr<Labels>)`.

Reply:
and the reture value between train_machine and fit_machine is different.

Reply:
yes it all needs to be refactored. My point was that maybe we can do one after the other. But it is OK to move on as you suggest here as well.
Keep in mind it is hard to change everything at once ...